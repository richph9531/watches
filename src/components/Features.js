import React from 'react';
import PropTypes from 'prop-types';

import classes from '../styles/Features.module.css';

const Features = props => {
  const featuresArray = props.data.data.featureList;
  const featureList = featuresArray.map((item, pos) => (
    <button
      key={pos}
      className={props.data.currentlySelectedFeature === item ? classes.SelectedFeatureItem : classes.FeatureItem }
      onClick={() => props.data.onFeatureSelection(pos)}
    >
    {item}
    </button>
  ));

  return (
    <div>
      <h3 className={classes.FeaturesHeading}>Features</h3>
      <div>{ featureList }</div>
     </div>
  );
};

Features.propTypes = {
  data: PropTypes.object,
  onFeatureSelection: PropTypes.object,
};

export default Features;
