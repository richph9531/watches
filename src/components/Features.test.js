import React from 'react';

import { render } from '@testing-library/react';
import Features from './Features';

test('Component renders correctly with Time selected', () => {
  const data = {
    data: {
      featureList: ['Time', 'Heart Rate'],
    },
    currentlySelectedFeature: 'Time',
  };
  const renderedElement = render(
  <Features
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Component renders correctly with Heart Rate selected', () => {
  const data = {
    data: {
      featureList: ['Time', 'Heart Rate'],
    },
    currentlySelectedFeature: 'Heart Rate',
  };
  const renderedElement = render(
  <Features
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Component renders correctly with an invalid value selected', () => {
  const data = {
    data: {
      featureList: ['Time', 'Heart Rate'],
    },
    currentlySelectedFeature: 'INVALID',
  };
  const renderedElement = render(
  <Features
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Component renders correctly with more values in the array', () => {
  const data = {
    data: {
      featureList: ['Time', 'Heart Rate', 'GPS'],
    },
    currentlySelectedFeature: 'GPS',
  };
  const renderedElement = render(
  <Features
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

