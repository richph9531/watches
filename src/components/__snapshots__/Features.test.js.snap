// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly with Heart Rate selected 1`] = `
<body>
  <div>
    <div>
      <h3
        class="FeaturesHeading"
      >
        Features
      </h3>
      <div>
        <button
          class="FeatureItem"
        >
          Time
        </button>
        <button
          class="SelectedFeatureItem"
        >
          Heart Rate
        </button>
      </div>
      <div>
        <button
          class="BuyNow"
        >
          Buy Now
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Component renders correctly with Time selected 1`] = `
<body>
  <div>
    <div>
      <h3
        class="FeaturesHeading"
      >
        Features
      </h3>
      <div>
        <button
          class="SelectedFeatureItem"
        >
          Time
        </button>
        <button
          class="FeatureItem"
        >
          Heart Rate
        </button>
      </div>
      <div>
        <button
          class="BuyNow"
        >
          Buy Now
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Component renders correctly with an invalid value selected 1`] = `
<body>
  <div>
    <div>
      <h3
        class="FeaturesHeading"
      >
        Features
      </h3>
      <div>
        <button
          class="FeatureItem"
        >
          Time
        </button>
        <button
          class="FeatureItem"
        >
          Heart Rate
        </button>
      </div>
      <div>
        <button
          class="BuyNow"
        >
          Buy Now
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Component renders correctly with more values in the array 1`] = `
<body>
  <div>
    <div>
      <h3
        class="FeaturesHeading"
      >
        Features
      </h3>
      <div>
        <button
          class="FeatureItem"
        >
          Time
        </button>
        <button
          class="FeatureItem"
        >
          Heart Rate
        </button>
        <button
          class="SelectedFeatureItem"
        >
          GPS
        </button>
      </div>
      <div>
        <button
          class="BuyNow"
        >
          Buy Now
        </button>
      </div>
    </div>
  </div>
</body>
`;
